container/writer.go

```diff
@@ -70,6 +72,14 @@ func NewWriter(writer io.Writer, codec Codec, recordsPerBlock int64) (*Writer, e
 		avroWriter.compressedWriter = avroWriter.blockBuffer
 	}
 
+	if len(schema) == 1 {
+		avroWriter.headerWritten = true
```

Comment:
headerWritten should be set to true only after checking that the err from writeHeader is nil - suggest moving this line down after line 80